src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
```

Comment:
The stateCopy is created outside of the loop, which means that it will be mutated across different actions. This will lead to incorrect results because each action should be applied to the state that was the result of the previous action, not the original state. You should create a new copy of the state inside the loop before applying each action.